### ISSUE TITLE
Skipped Test Fix

### ISSUE THREAD

**Issue by barnjamin** [cloudflare/cloudflare-typescript#2443]
### Confirm this is a Typescript library issue and not an underlying Cloudflare API issue

- [x] This is an issue with the Typescript library

### Describe the bug

There are a number of skipped tests in the `dns records` test section eg [here](https://github.com/cloudflare/cloudflare-typescript/blob/main/tests/api-resources/dns/records.test.ts#L14-L118).

I'm guessing they're skipped because the mock server operates only on the OpenAPI spec downloaded from the url in the `.stats.yml` file. There seems to be a bug(?) in the definition of at least the `components.schemas.dns-records_dns-record-post`. In the latest version I downloaded, a `data` field is marked as required. This field does not seem to _actually_ be required though, so the mock server is complaining about a non-issue.

Once I remove that field from the required array, I'm able to run my tests successfully. 

![Image](https://github.com/user-attachments/assets/ffe44645-036e-41e2-91e6-32b5d1fe8775)

--

I realize you did not intend for your CI stuff (like the mock server) to be part of the public API for this library so this is not a complaint about that, hopefully just a helpful tip to get the tests running again. 

### To Reproduce

Allow skipped tests to be run
Run tests -- see fail
Patch .yml (but its json??) file to remove `data` requirement
Run tests -- see success

### Code snippets

```Typescript

```

### OS

macOS

### Runtime version

Bun 1.2.4 

### Library version

v4.1.0

**Comment by barnjamin** [comment]
fwiw the same is true of the spec hosted in the `api-schemas` repo but I can't make an issue there

```sh
~ % curl -s "https://raw.githubusercontent.com/cloudflare/api-schemas/refs/heads/main/openapi.json" | jq '.components.schemas["dns-records_dns-record-post"]'
{
  "allOf": [
    {
      "$ref": "#/components/schemas/dns-records_dns-record"
    }
  ],
  "required": [
    "name",
    "type",
    "content",
    "data"
  ],
  "type": "object"
}
```

**Comment by jacobbednarz** [comment]
thanks for the issue however, we're aware of these and have internal tickets tracking the specifics to handle the OpenAPI issues.
